### PR TITLE
fix: check depth of embedded object and array

### PR DESCRIPTION
**Pull request by jgiovaresco**
**Issue**

https://gravitee.atlassian.net/browse/APIM-1614

**Description**

A small description of what you did in that PR.

**Additional context**

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `2.0.1-apim1614-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/policy/gravitee-policy-json-xml/2.0.1-apim1614-SNAPSHOT/gravitee-policy-json-xml-2.0.1-apim1614-SNAPSHOT.zip)

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -107,3 +107,7 @@ You can configure the policy with the following options:
 | Outgoing message cannot be transformed properly to XML
 
 |===
+
+=== Nested objects
+
+To limit the processing time in case of nested object, a default max depth of nested object has been defined to 1000. This default value can be overriden using the environment variable `gravitee_policy_json_xml_maxdepth`.
```
Comment (ytvnr) [inline README.adoc:113]: In XML-To-json policy we set it to 100, maybe we should keep 100 to be consistent.
by the way, a depth of 100 is already big, so it should be sufficient as a default value ?
wdyt ?

File: src/main/java/io/gravitee/policy/json2xml/transformer/JSONTokener.java [hunk 2]
```diff
@@ -50,6 +58,8 @@ of this software and associated documentation files (the "Software"), to deal
  */
 public class JSONTokener {
 
+    public static final int DEFAULT_MAX_DEPTH = 1000;
```
Comment (ytvnr) [inline src/main/java/io/gravitee/policy/json2xml/transformer/JSONTokener.java:61]: see previous comment

Comment (ytvnr) [reply src/main/java/io/gravitee/policy/json2xml/transformer/JSONTokener.java:61]: https://github.com/gravitee-io/gravitee-policy-xml-json/commit/295343e5c94898f6c6ced39b0a7e4dcfaf7cd651

File: README.adoc (hunk 1 above)
Comment (jgiovaresco) [reply README.adoc:113]: I've cherry picked the commit from `master` 🤔 
I can make the change but I will need to do it on `master` and `1.x` branches 🤷

**Review by ytvnr — APPROVED** [review]

**Pull request merged**